client/src/components/Button/index.tsx

```diff
@@ -1,11 +1,48 @@
-import { ReactNode } from 'react';
-
+type ColorType = 'main-blue' | 'white' | 'grey';
 interface ButtonProps {
-  children: ReactNode;
+  children: React.ReactNode;
+  title: string;
+  size: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
+  color: ColorType;
+  onClick: () => void;
+}
+
+function Button({ children, size, color, onClick }: ButtonProps) {
+  const buttonColor = getButtonColor(color);
+  const buttonSize = getButtonSize(size);
+  const className = `flex justify-center items-center rounded-[6px] ${buttonSize} ${buttonColor}`;
+
+  return (
+    <button className={className} onClick={onClick}>
+      {children}
+    </button>
+  );
+}
+
+function getButtonSize(size: string) {
+  switch (size) {
+    case 'xs':
+      return 'w-80px h-32px';
+    case 'sm':
+      return 'w-113px h-46px';
+    case 'md':
+      return 'w-140px h-50px';
+    case 'lg':
+      return 'w-150px h-46px';
+    case 'xl':
+      return 'w-244px h-57px';
+  }
 }
```

Comment:
이러한 방식에서는 일반적인 경우에 다음과 같이 사용하는 것으로 알고 있습니다! 특별히 이렇게 하신 이유가 있을까요?

```suggestion
const buttonSize = {
  xs : 'w-80px h-32px',
  sm : 'w-113px h-46px',
  md: 'w-140px h-50px',
  lg: 'w-150px h-46px',
  xl: 'w-244px h-57px'
}

// 이후 버튼 사용시
const className = `flex justify-center items-center rounded-[6px] ${buttonSize[size]} ${buttonColor[color]}`
```

Reply:
객체로 할 생각을 못했었네요! 객체로 하는 부분이 훨씬 명확하고 좋은것 같아요~ 반영하였습니다😃